test/Abstractions.Tests/TaskActivityTests.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Tests;
+
+public class TaskActivityTests
+{
+    [Theory]
+    [InlineData(typeof(ReferenceActivity), null)]
+    [InlineData(typeof(ReferenceActivity), "input")]
+    [InlineData(typeof(NullableReferenceActivity), null)]
+    [InlineData(typeof(NullableReferenceActivity), "input")]
+    [InlineData(typeof(ValueActivity), null)]
```

Comment:
Validating the experience of using a non-nullable value type as `TInput`. What happens when `null` is passed in to `ITaskActivity/Orchestrator.RunAsync(Context context, object? input)`? I opted to go with taking on the default value of the value type - and so I have to special case the return type to expect `0` (`default(int)`) as the result.

I _could_ instead fail here. As I can verify we got `null` and `typeof(TInput)IsValueType == true`. Would you prefer that behavior?

Reply:
Ah I see now. No, I think what you have now is the right behavior. I'd rather not fail in cases like this since (IMO) it just makes the API harder to use (not that users should generally be using these APIs directly...).

Reply:
I was undecided what was correct, but I agree with you. The motivating factor for me is that this way will keep the two calls to `RunAsync` with a `default` input identical in behavior.

``` CSharp
TaskOrchestrator<int>.RunAsync(context, default);
ITaskOrchestrator.RunAsync(context, default);
```